### PR TITLE
[Sprint 2] Select The Color of Cake Face

### PR DESCRIPTION
**Pull request by Tuanlvt**
Task : Logic Create Cake

### CODE REVIEW

File: app/src/main/java/com/fstyle/cakezy/screen/facecake/FaceCakePresenter.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+/**
+ * Listens to user actions from the UI ({@link FaceCakeActivity}), retrieves the data and updates
+ * the UI as required.
+ */
+final class FaceCakePresenter implements FaceCakeContract.Presenter {
+    private static final String TAG = FaceCakePresenter.class.getName();
+
+    private FaceCakeContract.ViewModel mViewModel;
+
+    public FaceCakePresenter() {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/cakezy/screen/facecake/FaceCakePresenter.java]: ERROR: Redundant 'public' modifier.

File: app/src/main/java/com/fstyle/cakezy/utils/ColorManager.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package com.fstyle.cakezy.utils;
+
+import android.graphics.Color;
+
+/**
+ * Created by Tuanlvt on 13/11/2017.
+ */
+
+public class ColorManager {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/cakezy/utils/ColorManager.java]: WARNING: Utility classes should not have a public or default constructor.

File: app/src/main/java/com/fstyle/cakezy/utils/Constant.java [hunk 3]
```diff
@@ -9,6 +9,16 @@
     public static final String END_POINT_URL = "http://google.com";
 
     private Constant() {
-        // No-op
+    }
+
+    public static class NumberIntegerAndNumberFloat {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/cakezy/utils/Constant.java]: WARNING: Missing a Javadoc comment.

File: app/src/main/java/com/fstyle/cakezy/utils/ColorManager.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package com.fstyle.cakezy.utils;
+
+import android.graphics.Color;
+import lombok.AllArgsConstructor;
+
+/**
+ * Created by Tuanlvt on 13/11/2017.
+ */
+@AllArgsConstructor
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/cakezy/utils/ColorManager.java]: WARNING: Utility classes should not have a public or default constructor.

File: app/src/main/java/com/fstyle/cakezy/utils/ColorManager.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
+package com.fstyle.cakezy.utils;
+
+import android.graphics.Color;
+
+/**
+ * Created by Tuanlvt on 13/11/2017.
+ */
+public class ColorManager {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/cakezy/utils/ColorManager.java]: WARNING: Utility classes should not have a public or default constructor.

File: app/src/main/java/com/fstyle/cakezy/utils/ColorManager.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
+package com.fstyle.cakezy.utils;
+
+import android.graphics.Color;
+
+/**
+ * Created by Tuanlvt on 13/11/2017.
+ */
+public final class ColorManager {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/cakezy/utils/ColorManager.java]: WARNING: Utility classes should not have a public or default constructor.

File: app/src/main/java/com/fstyle/cakezy/utils/ColorManager.java [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+    private static final String COLOR_PURPLE_100 = "#E1BEE7";
+    private static final String COLOR_PINK_50 = "#FCE4EC";
+    private static final String COLOR_PINK_100 = "#F8BBD0";
+    private static final String COLOR_BROWN_50 = "#EFEBE9";
+    private static final String COLOR_ORANGE_50 = "#FFF3E0";
+    private static final String COLOR_ORANGE_100 = "#FFE0B2";
+    private static final String COLOR_GREEN_100 = "#C8E6C9";
+
+    private ColorManager() {
+    }
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/cakezy/utils/ColorManager.java]: add // No-op

File: app/src/main/java/com/fstyle/cakezy/screen/facecake/FaceCakeModule.java [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    private Activity mActivity;
+
+    public FaceCakeModule(@NonNull Activity activity) {
+        this.mActivity = activity;
+    }
+
+    @ActivityScope
+    @Provides
+    public FaceCakeContract.ViewModel provideViewModel(FaceCakeContract.Presenter presenter) {
+        return new FaceCakeViewModel(mActivity, provideNavigator(), provideColorFaceAdapter(),
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/cakezy/screen/facecake/FaceCakeModule.java]: .

File: app/src/main/java/com/fstyle/cakezy/screen/facecake/FaceCakeViewModel.java [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+public class FaceCakeViewModel implements FaceCakeContract.ViewModel,
+        BaseRecyclerViewAdapter.OnRecyclerViewItemClickListener<Object> {
+
+    private Context mContext;
+    private Navigator mNavigator;
+    private FaceCakeContract.Presenter mPresenter;
+    private ColorFaceCakeAdapter mAdapter;
+    private ObservableField<ColorCake> mColorCakeObservableField = new ObservableField<>();
+
+    FaceCakeViewModel(Context context, Navigator navigator, ColorFaceCakeAdapter adapter,
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/cakezy/screen/facecake/FaceCakeViewModel.java]: pls remove context, navigator

File: app/src/main/java/com/fstyle/cakezy/screen/facecake/adapter/ColorFaceCakeAdapter.java [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+import com.fstyle.cakezy.databinding.ItemColorCakeBinding;
+import com.fstyle.cakezy.screen.BaseRecyclerViewAdapter;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.Setter;
+
+/**
+ * Created by Tuanlvt on 13/11/2017.
+ */
+@Setter
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/cakezy/screen/facecake/adapter/ColorFaceCakeAdapter.java]: remove

File: app/src/main/java/com/fstyle/cakezy/screen/facecake/adapter/ItemColorFaceCakeViewModel.java [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+            mItemClickListener;
+
+    ItemColorFaceCakeViewModel(ColorCake colorCake,
+            BaseRecyclerViewAdapter.OnRecyclerViewItemClickListener<Object> listener) {
+        mColorCake = colorCake;
+        mItemClickListener = listener;
+    }
+
+    public int getColorCake() {
+        return mColorCake.getMColorCake();
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/cakezy/screen/facecake/adapter/ItemColorFaceCakeViewModel.java]: remove m in Model

File: app/src/main/java/com/fstyle/cakezy/screen/choosestylecake/ChooseStyleCakeViewModel.java [hunk 12]
```diff
@@ -2,6 +2,7 @@
 
 import com.fstyle.cakezy.screen.decorationscake.DecorationsCakeActivity;
 import com.fstyle.cakezy.utils.navigator.Navigator;
+import javax.inject.Inject;
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/cakezy/screen/choosestylecake/ChooseStyleCakeViewModel.java]: ERROR: Unused import: javax.inject.Inject. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/fstyle/cakezy/screen/decorationscake/DecorationsCakeViewModel.java [hunk 13]
```diff
@@ -10,9 +10,11 @@
 import com.fstyle.cakezy.screen.BaseRecyclerViewAdapter;
 import com.fstyle.cakezy.screen.decorationscake.adaptercandle.CandleAdapter;
 import com.fstyle.cakezy.screen.decorationscake.adaptercream.CreamAdapter;
+import com.fstyle.cakezy.screen.facecake.FaceCakeActivity;
 import com.fstyle.cakezy.utils.navigator.Navigator;
 import java.util.ArrayList;
 import java.util.List;
+import javax.inject.Inject;
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/cakezy/screen/decorationscake/DecorationsCakeViewModel.java]: ERROR: Unused import: javax.inject.Inject. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/fstyle/cakezy/data/model/ColorCake.java [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+import lombok.Setter;
+
+/**
+ * Created by Tuanlvt on 14/11/2017.
+ */
+@Getter
+@Setter
+@AllArgsConstructor
+public class ColorCake extends BaseModel {
+    private int colorCake;
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/cakezy/data/model/ColorCake.java]: WARNING: [AvoidFieldNameMatchingTypeName] It is somewhat confusing to have a field name matching the declaring class name
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#AvoidFieldNameMatchingTypeName

File: app/src/main/java/com/fstyle/cakezy/data/model/CreamCake.java [hunk 15]
```diff
@@ -10,5 +10,5 @@
 @AllArgsConstructor
 @Getter
 public class CreamCake extends BaseModel {
-    private Bitmap mCreamCake;
+    private Bitmap creamCake;
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/cakezy/data/model/CreamCake.java]: WARNING: [AvoidFieldNameMatchingTypeName] It is somewhat confusing to have a field name matching the declaring class name
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#AvoidFieldNameMatchingTypeName

File: app/src/main/java/com/fstyle/cakezy/screen/choosestylecake/ChooseStyleCakeViewModel.java (hunk 12 above)
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/cakezy/screen/choosestylecake/ChooseStyleCakeViewModel.java]: WARNING: [UnusedImports] Avoid unused imports such as 'javax.inject.Inject'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/fstyle/cakezy/screen/decorationscake/DecorationsCakeViewModel.java (hunk 13 above)
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/cakezy/screen/decorationscake/DecorationsCakeViewModel.java]: WARNING: [UnusedImports] Avoid unused imports such as 'javax.inject.Inject'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/fstyle/cakezy/screen/facecake/FaceCakeActivity.java [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+
+        DaggerFaceCakeComponent.builder()
+                .appComponent(((MainApplication) getApplication()).getAppComponent())
+                .faceCakeModule(new FaceCakeModule(this))
+                .build()
+                .inject(this);
+
+        ActivityFaceCakeBinding binding =
+                DataBindingUtil.setContentView(this, R.layout.activity_face_cake);
+        binding.setViewModel((FaceCakeViewModel) mViewModel);
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/cakezy/screen/facecake/FaceCakeActivity.java]: ERROR: [PMD-CPD] 37 lines duplicated.
app/src/main/java/com/fstyle/cakezy/screen/decorationscake/DecorationsCakeActivity.java

File: app/src/main/java/com/fstyle/cakezy/screen/facecake/FaceCakeActivity.java [hunk 17]
```diff
@@ -0,0 +1,71 @@
...
+        setUpActionBar(true, R.string.decorations_face_cake);
+
+        DaggerFaceCakeComponent.builder()
+                .appComponent(((MainApplication) getApplication()).getAppComponent())
+                .faceCakeModule(new FaceCakeModule(this))
+                .build()
+                .inject(this);
+
+        ActivityFaceCakeBinding binding =
+                DataBindingUtil.setContentView(this, R.layout.activity_face_cake);
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/cakezy/screen/facecake/FaceCakeActivity.java]: ERROR: [PMD-CPD] 38 lines duplicated.
app/src/main/java/com/fstyle/cakezy/screen/decorationscake/DecorationsCakeActivity.java

Comment (Tuanlvt) [reply app/src/main/java/com/fstyle/cakezy/screen/facecake/FaceCakeActivity.java]: edited

**Pull request merged**